### PR TITLE
1.30: Users API Updates

### PR DESCRIPTION
**Pull request by bguyza**
### Updates
* Request objects update:
  * `disableDomainBlock` added
  * `applicationId` added for email templates.
 * `DisplayableRawLogin` updated to include `Location`.  Response Body and example JSON updated for `api/user/recent-login`.
* (Not User API related but made it into this commit anyway) New roles added for `acl_manager` and `acl_deleter

### CODE REVIEW

File: site/docs/src/json/users/create-request.json [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
+  "application": "368994e6-ba8a-49b5-8446-7b99ab1fffdf",
```
Comment (bguyza) [inline site/docs/src/json/users/create-request.json:2]:
```suggestion
  "applicationId": "368994e6-ba8a-49b5-8446-7b99ab1fffdf",
```

File: site/docs/v1/tech/apis/_user-registration-combined-request-body.adoc [hunk 2]
```diff
@@ -5,6 +5,9 @@ This request requires that you specify both the User object and the User Registr
 :set_password_field_config_blurb: configured in the Application, or, if that is not configured, the same field in the Tenant
 
 [.api]
+[field]#disableDomainBlock# [type]#[Boolean]# [optional]#Optional# [default]#defaults to `false`# [since]#Available since 1.30.0#::
+If `true`, blocked domains that have been configured on the tenant will be ignored when creating or updating a User.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/_user-registration-combined-request-body.adoc:9]: Suggestion:


> A tenant has the option to configure one or more email domains to be blocked in order to restrict email domains during user create or update.
> 
> Setting this property equal to `true` will override the tenant configuration. See [field]#tenant.registrationConfiguration.blockdDomains# in the Tenant API.

File: site/docs/v1/tech/apis/_user-request-body.adoc [hunk 3]
```diff
@@ -7,6 +7,12 @@ You must specify either the **email** or the **username** or both for the User.
 
 
 [.api]
+[field]#applicationId# [type]#[UUID]# [optional]#Optional# [since]#Available since 1.30.0#::
+The Id of the Application that is used for email templates.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/_user-request-body.adoc:11]: Suggestion:

> An optional Application Id. This value is optional, and when provided will be used to resolve an application specific email template if you have configured transactional emails such as setup password, email verification and others. 
> 
> If not provided, only the tenant configuration will be used when resolving email templates.

File: site/docs/v1/tech/apis/users.adoc [hunk 4]
```diff
@@ -762,6 +762,24 @@ The link:/docs/v1/tech/reference/data-types/#instants[instant] this login occurr
...
+
+[field]#logins``[x]``.location.longitude# [type]#[Double]# [since]#Available since 1.30.0#::
+The longitude where the login request originated.
+
+[field]#logins``[x]``.location.region# [type]#[String]# [since]#Available since 1.30.0#::
+The geographic location where the login request originated.
+
+[field]#logins``[x]``.location.zipcode# [type]#[String]# [since]#Available since 1.30.0#::
+The zipcode where the login request originated.
+
```
Comment (robotdan) [inline site/docs/v1/tech/apis/users.adoc:782]: These fields are only returned when you are using the Threat Detection feature. I think we have some wording in other places to designate fields or features that are only included with a particular license or edition.

File: site/docs/v1/tech/apis/users.adoc [hunk 5]
```diff
@@ -762,6 +762,24 @@ The link:/docs/v1/tech/reference/data-types/#instants[instant] this login occurr
 [field]#logins``[x]``.ipAddress# [type]#[String]#::
 The IP address if provided during the login request.
 
+[field]#logins``[x]``.location.city# [type]#[String]# [since]#Available since 1.30.0#::
+The city where the login request originated.
+
+[field]#logins``[x]``.location.country# [type]#[String]# [since]#Available since 1.30.0#::
+The country where the login request originated.
+
+[field]#logins``[x]``.location.latitude# [type]#[Double]# [since]#Available since 1.30.0#::
```
Comment (robotdan) [inline site/docs/v1/tech/apis/users.adoc:771]: Above in the example JSON response you added `displayString` but omitted it here. It is technically a synthetic field, but it looks like. I didn't add  `@JSONIgnore` on it - so it is coming back in the response. This is probably ok, but if we are going to document it in the JSON we should probably add it here as well.

File: site/docs/v1/tech/core-concepts/roles.adoc [hunk 6]
```diff
@@ -89,6 +89,8 @@ Below are all the roles available in FusionAuth. Please note the additional expl
 | Name | Id | Description
 
 | `admin` | `631ecd9d-8d40-4c13-8277-80cedb8236e2` | Can manage everything, including creating new users with administrator privileges.
+| `acl_manager` | `631ecd9d-8d40-4c13-8277-80cedb823712` | Can add and edit IP access control lists. Available since 1.30.
+| `acl_deleter` | `631ecd9d-8d40-4c13-8277-80cedb823711` | Can delete IP access control lists. Available since 1.30.
```
Comment (robotdan) [inline site/docs/v1/tech/core-concepts/roles.adoc:93]: Use `1.30.0`, probably update the other usages as well to include the `.0`

**Review by robotdan — COMMENTED** [review]
Comments for `applicationId` and `disableDomainBlock` apply to all usages.

File: site/docs/v1/tech/apis/users.adoc (hunk 4 above)
Comment (mooreds) [reply site/docs/v1/tech/apis/users.adoc:782]: You want `docs/v1/tech/shared/_premium-edition-blurb-api.adoc`

**Pull request merged**